src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -226,19 +240,32 @@ public void periodic() {
                 } else {
                         updateRotationTarget = true;
                 }
+
+                if(Robot.isSimulation()) {
+                        if(!DriverStationSim.getAutonomous()) {
+                                states = kinematics.toSwerveModuleStates(chassisSpeeds);
+                        }
+                        positionMeters[0] += states[0].speedMetersPerSecond * (Timer.getFPGATimestamp() - lastUpdateTime);
+                        positionMeters[1] += states[1].speedMetersPerSecond * (Timer.getFPGATimestamp() - lastUpdateTime);
+                        positionMeters[2] += states[2].speedMetersPerSecond * (Timer.getFPGATimestamp() - lastUpdateTime);
+                        positionMeters[3] += states[3].speedMetersPerSecond * (Timer.getFPGATimestamp() - lastUpdateTime);
+                        simRotation += chassisSpeeds.omegaRadiansPerSecond * (Timer.getFPGATimestamp() - lastUpdateTime);
```

Comment:
For this to work properly in all cases (auto when chassisSpeeds is undefined) we need something like the following:

```
double radiansPerSecond = kinematics.toChassisSpeeds(states).omegaRadiansPerSecond;
simRotation += radiansPerSecond * (Timer.getFPGATimestamp() - lastUpdateTime); 
```

Reply:
I did a little testing of this with our current auto path and did not see any rotation but realized that PathPlanner by default for swerve only uses translation. If you go to settings for PathPlanner you can enable "holonomic" mode which allows you to specify the holonomic angle (swerve chassis rotation) and after making a path that had non-zero holonoimc angles you can see the bot rotate on the field widget in auto with the above change. Give it a try! it was pretty cool to see the simulated robot strafe around like it did in PathPlanner